include/zephyr/drivers/adc.h

```diff
@@ -704,6 +712,7 @@ __subsystem struct adc_driver_api {
 #ifdef CONFIG_ADC_ASYNC
 	adc_api_read_async    read_async;
 #endif
+	adc_api_get_ref_internal get_ref_internal;
```

Comment:
This is quit drastic, maybe not allowing the `adc_emul` driver to adjust the `ref_internal` value is better/easier.